lib/usb-connection.js

```diff
@@ -144,13 +147,22 @@ USB.Connection.prototype.open = function() {
           if (err) {
             reject(err);
           } else {
-            resolve(data);
+            this.serialNumber = data;
+            resolve();
           }
         });
       });
-    })
-    .then((data) => {
-      this.serialNumber = data;
+    });
+
+  if (altSetting === 1) {
```

Comment:
I don't understand why this logic is done outside the Promise? `this.setAltSetting(altSetting)` could be returned instead of saved to a variable, and this logic could be done inside the next `.then()` statement:
 
from line 156
```js
...
.then(() => {
    if (altSetting === 1) {
        return Promise.resolve(this);
    } else {
        // everything else past line 167
        ...
    }
});
```

This should reduce some of the complexity of passing around a Promise, as well as ensure the error case is handled in one place.